File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode pointer1 = l1;
        ListNode pointer2 = l2;

        ListNode dummy = new ListNode(0);
        ListNode pointer3 = dummy;
        int carry = 0;

        while (pointer1 != null && pointer2 != null) {
            int a = pointer1.val;
            int b = pointer2.val;
            int c = a + b + carry;
            carry = c / 10;
            pointer3.next = new ListNode(c % 10);
            pointer3 = pointer3.next;

            pointer1 = pointer1.next;
            pointer2 = pointer2.next;
        }

        while (pointer1 != null) {
            int val = pointer1.val;
            int c = val + carry;
            carry = c / 10;
            pointer3.next = new ListNode(c % 10);
            pointer3 = pointer3.next;
            pointer1 = pointer1.next;
        }

        while (pointer2 != null) {
            int val = pointer2.val;
            int c = val + carry;
            carry = c / 10;
            pointer3.next = new ListNode(c % 10);
            pointer3 = pointer3.next;
            pointer2 = pointer2.next;
        }

        if (carry > 0) {
            pointer3.next = new ListNode(carry);
        }

        return dummy.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:


        pointer1 = l1
        pointer2 = l2

        dummy = ListNode(0)
        pointer3 = dummy
        carry = 0
        while pointer1 and pointer2:

            a = pointer1.val
            b = pointer2.val
            c = a+b
            if carry>0:
                c = c+carry
                carry = 0
            if c<10:
                temp = ListNode(c)
                pointer3.next = temp
                pointer3 = pointer3.next
            else:
                right = c%10
                carry = c//10
                temp = ListNode(right)
                pointer3.next = temp
                pointer3 = pointer3.next

            pointer1 = pointer1.next
            pointer2 = pointer2.next

        while pointer1:
            print(carry)

            val = pointer1.val 
            c = val
            if carry>0:
                c = c+carry
                carry = 0
            if c<10:
                temp = ListNode(c)
                pointer3.next = temp
                pointer3 = pointer3.next
            else:
                right = c%10
                carry = c//10
                temp = ListNode(right)
                pointer3.next = temp
                pointer3 = pointer3.next
            pointer1 = pointer1.next

        while pointer2:
            val = pointer2.val 
            c = val
            if carry>0:
                c = c+carry
                carry = 0
            if c<10:
                temp = ListNode(c)
                pointer3.next = temp
                pointer3 = pointer3.next
            else:
                right = c%10
                carry = c//10
                temp = ListNode(right)
                pointer3.next = temp
                pointer3 = pointer3.next

            pointer2 = pointer2.next
        if carry:
            temp = ListNode(carry)
            pointer3.next = temp
        return dummy.next
